### PR TITLE
lecture 1 solutions

### PR DESCRIPTION
**Pull request by dupreehkuda**
### Solutions for Lecture 1

cc @chshersh

Thank you in advance for checking!
As I understand I accomplished the task not in 'haskell-way', so correct me pls if I'm wrong.

### CODE REVIEW

**Comment by chshersh** [comment]
Hi @dupreehkuda 👋🏻 

I see that you mentioned my username but you already merged the PR. In that case, would you still like to get my feedback on your solutions?

I might not be able to provide it quickly as it's currently Christmas and New Year holidays but I'll try to do it later 🙂

**Comment by dupreehkuda** [comment]
Oh hi @chshersh! Yeah, I thought to move on and closed it, but still I'll be grateful if you check it;) Oh and by the way... С наступающим!

File: src/Lecture1.hs [hunk 1]
```diff
@@ -67,7 +70,8 @@ sumOfSquares x y = error "TODO!"
 
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = if n < 0 then mod (negate n) 10 else mod n 10
```
Comment (chshersh) [inline src/Lecture1.hs:74]: That's correct! 💯 
You can simplify the implementation a bit using the standard function `abs` that returns an absolute value of a number:

```haskell
λ: abs 7
7
λ: abs (-7)
7
```

File: src/Lecture1.hs [hunk 2]
```diff
@@ -93,12 +98,20 @@ the end (including).
...
+
+
 This function can accept negative start and end position. Negative
 start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+startFormat :: Int -> Int
+startFormat start = if start < 0 then 0 else start
```
Comment (chshersh) [inline src/Lecture1.hs:111]: You can write this function shorter by using standard Haskell function `max`:

```haskell
startFormat start = max 0 start
```

File: src/Lecture1.hs [hunk 3]
```diff
@@ -108,7 +121,12 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+
+toInt :: [String] -> [Int]
+toInt = map read
+
+strSum :: [Char] -> Int
+strSum str = sum (toInt (words str))
```
Comment (chshersh) [inline src/Lecture1.hs:129]: As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can inline the definition of `toInt` directly here:

```suggestion
strSum str = sum (map read (words str))
```

**Review by chshersh — COMMENTED** [review]
Congrats! The first homework is complete 👏🏼 
Hope you enjoyed the course so far. Keep up the great work, and feel free to ask any questions if you have 👌🏼

С Новым Годом! 🥳 🍾 🎄

**Pull request merged**